### PR TITLE
fix(TDI-38901):Can't retrieve the begining or end of Azure queue name

### PR DESCRIPTION
**Pull request by pyzhou-talend**
with numberic

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-38901

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/426/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/426/)

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/wizard/AzureStorageComponentListProperties.java [hunk 1]
```diff
@@ -135,6 +135,9 @@ public ValidationResult afterFormFinishTable(Repository<Properties> repo) throws
             for (NamedThing nl : selectedContainerNames.getValue()) {
```
Comment (mbasiuk-talend) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/wizard/AzureStorageComponentListProperties.java]: It seems, that this 3 "for" loops have almost the same logic inside. Is it possible to refactor it? (For instance: create a new method that will handle all cases)

**Comment by mbasiuk-talend** [comment]
Is there any tests for this issue?

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/wizard/AzureStorageComponentListProperties.java (hunk 1 above)
Comment (pyzhou-talend) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/wizard/AzureStorageComponentListProperties.java]: It's difficult to combine these three for loops into one. There's too many differences between them. Refactor won't reduce much redundancy and will make the code hard to read.

**Comment by pyzhou-talend** [comment]
Junit Test have been added. Thanks for your help.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/439/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/439/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/440/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/440/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/476/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/476/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/552/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/552/)

**Pull request merged**